script.js

```diff
@@ -1,4 +1,137 @@
+  `You lost.\<br>\ System's ${iconForNoun(b)} beats your ${iconForNoun(
+    a
+  )}.\<br>\You: ${c}\<br>\System: ${d}.\<br>\Winning so far is ${e}.`;
+const koreanDrawMessage = (a, b, c, d, e) =>
+  `${drawMessage(a, b, c, d, e)}\<br>\ Ultimate winner is ${e}.`;
+const koreanWinMessage = (a, b, c, d, e) =>
+  `${winMessage(a, b, c, d, e)}\<br>\ You shout "muk-jji-ppa!".`;
+const koreanLoseMessage = (a, b, c, d, e) =>
+  `${loseMessage(a, b, c, d, e)}\<br>\ System shouts "muk-jji-ppa!".`;
+
+// All invalid cases of user's input
+const invalidCases = (a) =>
+  a.toLowerCase() !== "paper" &&
+  a.toLowerCase() !== "rock" &&
+  a.toLowerCase() !== "scissors";
+// All cases where user wins
+const winCases = (b, c) =>
+  (b.toLowerCase() === "paper" && c === "rock") ||
+  (b.toLowerCase() === "rock" && c === "scissors") ||
+  (b.toLowerCase() === "scissors" && c === "paper");
+// All cases where both parties choose the same object.
+const drawCases = (d, e) => d.toLowerCase() === e;
+
+// normal reverse korean mode game play
+const normalReversedKoreanGamePlay = (userInput, isReversed, isKorean) => {
+  let showMeHand = getRandomHand();
+  let winner;
+  //If invalid input , return feedback.
+  if (invalidCases(userInput)) {
+    return invalidMessage(userInput);
+  }
+  //If both parties choose the same object, it's a draw.
+  else if (drawCases(userInput, showMeHand)) {
+    winner = winnerOfGame();
+    //return ultimate winner declaration when it is korean mode
+    return isKorean
+      ? koreanDrawMessage(userInput, showMeHand, userWin, systemWin, winner)
+      : drawMessage(userInput, showMeHand, userWin, systemWin, winner);
+  }
+  // User wins.
+  else if (isReversed ^ winCases(userInput, showMeHand)) {
+    //increment userWin score
+    userWin += 1;
+    winner = winnerOfGame();
+    return isKorean
+      ? koreanWinMessage(userInput, showMeHand, userWin, systemWin, winner)
+      : winMessage(userInput, showMeHand, userWin, systemWin, winner);
+  }
+  // System wins.
+  else if (!(isReversed ^ winCases(userInput, showMeHand))) {
+    //increment systemWin score
+    systemWin += 1;
+    winner = winnerOfGame();
+    return isKorean
+      ? koreanLoseMessage(userInput, showMeHand, userWin, systemWin, winner)
+      : loseMessage(userInput, showMeHand, userWin, systemWin, winner);
+  }
+};
+
+//track scores
+let userWin = 0;
+let systemWin = 0;
+//track winner
+const winnerOfGame = () =>
+  systemWin > userWin ? "System" : systemWin < userWin ? "You" : "Nobody";
+//reset Game when switch mode
+const resetGame = () => {
+  userWin = 0;
+  systemWin = 0;
+};
+//define default game mode as pending for username
+let userName = "";
+let gameMode = "pending for username";
+
+// Main game
+const main = (input) => {
+  // waiting for user to enter username
+  if (gameMode == "pending for username") {
+    userName = input;
+    gameMode = "normal";
+    return `Welcome ${userName}.\<br>\You may choose to play normal, reverse, or korean game.`;
```

Comment:
There is something know as [SCREAM_CASE constants ](https://www.freecodecamp.org/news/when-to-capitalize-your-javascript-constants-4fabc0a4a4c4/)

I can touch more on this as the days go by, but it's simply another way to name variables that remain unchanged through out the program eg.
```
var STONE = "stone"
var SCISSORS = "scissors"
var PAPER = "paper"
```